src/Domain/Heroes/PowerLevelUpdatedEvent.cs

```diff
@@ -1,20 +1,23 @@
 ﻿using Ardalis.GuardClauses;
 using SSW.CleanArchitecture.Domain.Common.Base;
+using SSW.CleanArchitecture.Domain.Teams;
 
 namespace SSW.CleanArchitecture.Domain.Heroes;
 
 public record PowerLevelUpdatedEvent : DomainEvent
 {
     public HeroId Id { get; }
-    public string Name { get; }
-    public int PowerLevel { get; }
+    public TeamId? TeamId { get; }
```

Comment:
Why not just store the Hero?

Reply:
To capture the exact state of the hero when the changes were raised.  In our example it doesn't matter too much, but in other examples if many domain events were fired we would want them to be unique and NOT just point to the latest version of the Hero (which would all be the same)